File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int count = 0;
        int n = s.size();
        if (n == 0) {
            return 0;
        }
        vector<vector<bool>> dp(n, vector<bool>(n,false));
        for (int i = 0; i < n; i++) {
            dp[i][i] = true;
            count++;
        }
        for (int len = 2; len <= n; len++) {
            for (int i = 0; i <= n - len; i++) {
                int j = i + len - 1;
                if (s[i] == s[j]) {
                    if (len == 2 || dp[i + 1][j - 1]) {
                        dp[i][j] = true;
                        count++;
                    }
                }
            }
        }
        return count;
    }
};

File: Data Structures & Algorithms/jump-game-ii/submission-0.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        count = 0
        can_reach = 0
        max_dist = 0

        for i in range(len(nums) - 1):
            can_reach = max(can_reach, i + nums[i])
            if i == max_dist:
                count += 1
                max_dist = can_reach

        return count